//-------------------------------------------------------------------------------
// Name:        nmea_messages.proto
// Purpose:     protobuf and grpc for navigation servers nmea data streams
//
// Author:      Laurent Carré
//
// Created:     23/10/2023
// Copyright:   (c) Laurent Carré Sterwen Technology 2021-2024
// Licence:     Eclipse Public License 2.0
//-------------------------------------------------------------------------------

syntax = "proto3";

import "nmea2000.proto";
import "nmea0183.proto";

message nmea_msg {
  oneof Message {
    nmea2000pb N2K_msg = 1;
    nmea0183pb N0183_msg = 2;
  }
  uint32 msg_id=3;
}

message server_resp {
  uint32 reportCode = 1;
  string status = 2;
}

message server_cmd {
  string cmd=1;
}